### DISCUSSION TITLE
Access Stylistic-sets only for specific mathematical symbols

### DISCUSSION THREAD

**Discussion by yarakos95** [typst/typst#3212] [Q&A]
### Access Stylistic-sets

By configuring as follows, you can access ss03 of `U+210F` of STIX Two Math.

```typst
#show math.equation: set text(font: "STIX Two Math")
#let hbar = math.class(
  "normal",
  text(font: ("STIX Two Math", ""), stylistic-set: 03)[#math.planck.reduce]
)
```

![math-1](https://github.com/typst/typst/assets/75878492/01f2f588-f4e1-4a68-bf5a-4aabdb88ecf0)

Then `planck.reduce` and `hbar` will have different glyph shapes.

A strange thing is that the font specified by `text` has to be an array type. Stylistic-sets don't function properly if the `font` is a string type. I'm not sure if this issue with the string type should be reported.

---

Additionally, to access ss01, I configured:

```typst
#show math.equation: set text(font: "STIX Two Math")
#let scr(a) = math.class(
  "normal",
  text(font: ("STIX Two Math", ""), stylistic-set: 01)[#math.cal(a)]
)
```

![math-2](https://github.com/typst/typst/assets/75878492/eca8fb91-26e6-40b9-bfbe-2e6041016943)

Then `cal(A)` and `scr(A)` will have different glyph shapes. Similarly, `font` must be an array type.

### Problem

However, you cannot change it to `bold` using the above specification method.

![math-3](https://github.com/typst/typst/assets/75878492/f65640bc-bd5b-4585-88d2-d6f9eff3e6eb)

Is there any solution for this?

If you look at [wakamaifondue.com][], you can see that STIX Two Math has `scr` in bold.

[wakamaifondue.com]: https://wakamaifondue.com/ "Wakamai Fondue, the tool that answers the question “what can my font do?”"

![image](https://github.com/typst/typst/assets/75878492/b4f99b23-eb61-416f-ac00-a5cb91fab0d4)

A possible workaround is to declare `boldscr` like this: However, this may not be a very good idea.

```typst
#show math.equation: set text(font: "STIX Two Math")
#let scr(a) = math.class(
  "normal",
  text(font: ("STIX Two Math", ""), stylistic-set: 01)[#math.bold(math.cal(a))]
)
```

### Version

```
Typst Web Application
Deployed on 2024-01-09T22:45:20.076Z
Copyright © 2022—2024 Typst GmbH
Typst compiler version: 70ca0d257bb4ba927f63260e20443f244e0bb58c

Made in Berlin
```

**Comment by xkevio** [comment]
I don't quite know why it works when the font is specified as an array but the reason it doesn't work with just the font specified is because the show-set rule seems to override all future invocations of `text` in an equation with the one set in the rule. Meaning, that the `stylistic-set` parameter is set to `none` as the default.

I can't think of a pretty workaround unfortunately until `revoke`-rules exist.

**Reply by yarakos95** [reply]
@xkevio, thank you fou your reply.

Is `revoke`-rule what is being discussed here?; https://github.com/typst/typst/issues/420
I would like to keep an eye on how this issue will be implemented in the future. 👀

**Comment by laurmaedje** [comment]
I think the font override helps because of a hack in math layout. It forces the normal text layout pipeline, which properly handles stylistic-set. Without it, currently the math-specific text layout kicks in, which handles math subsitutions, but apparently doesn't handle stylistic sets correctly.

This, for example, also works because the `box` forces normal text layout:
```typst
#show math.equation: set text(font: "STIX Two Math")
#let hbar = math.class(
  "normal",
  box(text(stylistic-set: 03)[#math.planck.reduce])
)

$ hbar $
```

**Reply by yarakos95** [reply]
@laurmaedje, thank you for your reply.

> It forces the normal text layout pipeline, which properly handles stylistic-set. Without it, currently the math-specific text layout kicks in, which handles math subsitutions, but apparently doesn't handle stylistic sets correctly.

It was not understandable for me. This is because my understanding of Typst is shallow, sorry. Could you think of any specific situations where the example I presented would be inconvenient?

Also, even in the example using `box`, it is not possible to change `bold`. Is there any solution for this?

**Reply by laurmaedje** [reply]
It's just buggy. When `bold` doesn't work, you can also try `strong` (which is more or less the non-math counterpart).
